virtwho/virt/vdsm/vdsm.py

```diff
@@ -107,46 +112,120 @@ def _get_local_vds_name(self, trusted_store_path):
         out, err = p.communicate()
         if p.returncode != 0:
             return '0'
-        return re.search('/CN=([^/$\n]+)', out).group(1)
+        return re.search('CN *= *([^,/$\n]+)', out).group(1)
+
+    def _get_addr(self):
+        return self._get_local_vds_name(self.trust_store_path)
 
-    def _secure_connect(self):
-        addr = self._get_local_vds_name(self.trust_store_path)
+    @staticmethod
+    def _need_m2crypto():
+        """Determine whether M2Crypto usage is necessary.
 
-        from M2Crypto.m2xmlrpclib import SSL_Transport
-        from M2Crypto import SSL
+        If the python version supports SSL contexts and allows specifying them in xmlrpc clients, we do not need
+        M2Crypto.
+        """
+        if not hasattr(ssl, 'create_default_context'):
+            return False
+        try:
+            xmlrpclib.ServerProxy('https://localhost', context=ssl.create_default_context())
+            return True
+        except TypeError:
+            return False
 
+    def _create_ssl_context(self):
         key_file = self.trust_store_path + '/keys/vdsmkey.pem'
         cert_file = self.trust_store_path + '/certs/vdsmcert.pem'
         ca_cert = self.trust_store_path + '/certs/cacert.pem'
 
-        ctx = SSL.Context()
+        if self._need_m2crypto():
+            ctx = ssl.create_default_context(capath=ca_cert)
+            ctx.load_cert_chain(cert_file, key_file)
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
+        else:
+            from M2Crypto import SSL
 
-        ctx.set_verify(SSL.verify_peer | SSL.verify_fail_if_no_peer_cert, 16)
-        ctx.load_verify_locations(ca_cert)
-        ctx.load_cert(cert_file, key_file)
+            ctx = SSL.Context()
 
-        return xmlrpclib.Server('https://%s:%s' % (addr, self.management_port), SSL_Transport(ctx))
+            ctx.set_verify(SSL.verify_peer | SSL.verify_fail_if_no_peer_cert, 16)
+            ctx.load_verify_locations(ca_cert)
+            ctx.load_cert(cert_file, key_file)
+        return ctx
 
     def connect(self):
         if self.trust_store_path:
-            try:
-                self.server = self._secure_connect()
+            ssl_context = self._create_ssl_context()
+            addr = self._get_local_vds_name(self.trust_store_path)
```

Comment:
It seems that `addr` is get only from certificate and value of `server` option from configuration file is completely ignored. This logic was present in old xmlrpc code too. @cnsnyder, I don' think that this is expected behavior. Thoughts?